splice_machine/src/main/java/com/splicemachine/derby/impl/sql/compile/MergeJoinStrategy.java

```diff
@@ -227,30 +227,30 @@ private boolean mergeable(RowOrdering outerRowOrdering,
                 assert relop instanceof BinaryRelationalOperatorNode:
                         "Programmer error: RelationalOperator of type "+ relop.getClass()+" detected";
                 BinaryRelationalOperatorNode bron = (BinaryRelationalOperatorNode)relop;
-                ColumnReference innerColumn=relop.getColumnOperand(innerTable);
-                ColumnReference outerColumn=getOuterColumn(bron,innerColumn);
-                if (innerColumn == null || outerColumn == null) continue;
-                int innerColumnNumber = innerColumn.getColumnNumber();
-                if(innerColumnNumber==innerColumnPosition){
-                    innerColumns.set(i);
-                    int outerTableNum=outerColumn.getTableNumber();
-                    int outerColNum=outerColumn.getColumnNumber();
-                    if(ascending){
-                        int outerPos = outerRowOrdering.orderedPositionForColumn(RowOrdering.ASCENDING,outerTableNum,outerColNum);
-                        if(outerPos>=0){
-                            outerColumns.set(outerPos);
-                            innerToOuterJoinColumnMap[i] = outerPos;
+                if (bron.getOperator() == RelationalOperator.EQUALS_RELOP) {
```

Comment:
Are we forbidding any valid merge joins this way? If the last predicate (on the last index column) is an inequality we could still use merge join, right?

Reply:
Yes, @dgomezferro If all the leading index columns have either equality condition or are equal to a constant value, merge join can still be picked up even if the last index column has an inequality condition.
The 4th test case in MergeJoinIT.testMergeFeasiblityWithNonEqualityJoinConditionsOnKeys() is such an example. Thanks.

Reply:
@yxia92  I see, thanks!